presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -98,6 +98,20 @@ default boolean schemaExists(ConnectorSession session, String schemaName)
 
     ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle);
 
+    default ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession session, ConnectorTableLayoutHandle tableLayoutHandle, ConnectorPartitioningHandle partitioningHandle)
```

Comment:
This surely missing a comment.

Reply:
Maybe getPartitionedTableLayoutHandle? Maybe it could return Optional, then I suspect that below method would not need to be necessary.

Reply:
I added comment.

I don't think returning an Optional would be helpful in removing the other method.